frontend/app/src/MetricsManager.ts

```diff
@@ -240,4 +291,54 @@ export class MetricsManager {
     }
     return {}
   }
+
+  // Get context data for events
+  private getContextData(): Record<string, unknown> {
+    return {
+      contextPageUrl: window.location.href,
+      contextPageTitle: document.title,
+      contextPagePath: window.location.pathname,
+      contextPageReferrer: document.referrer,
+      contextPageSearch: window.location.search,
+      contextLocale:
+        // @ts-expect-error
+        window.navigator.userLanguage || window.navigator.language,
+      contextUserAgent: window.navigator.userAgent,
+    }
+  }
+
+  /**
+   * Get/Create user's anonymous ID
+   * Checks if existing in cookie or localStorage, otherwise generates
+   * a new UUID and stores it in both.
+   */
+  private getAnonymousId(): void {
+    // If metrics disabled, anonymous ID unnecessary
+    if (!this.actuallySendMetrics) return
+
+    const isLocalStoreAvailable = localStorageAvailable()
+    const anonymousIdCookie = getCookie("ajs_anonymous_id")
+    const anonymousIdLocalStorage = isLocalStoreAvailable
+      ? localStorage.getItem("ajs_anonymous_id")
```

Comment:
nit: the `ajs_anonymous_id` could potentially be put into a shared constant since it is used in quite a few places.